docs/data/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,214 @@
+    if (isNonRetryErrorCase(error)) {
+      // ...do other error handling...
+      return;
+    }
+
+    // the tx no longer has a chance of making it into a ledger
+    if (Date.now() >= expiration) {
+      return new Error("The transaction timed out.");
+    }
+
+    timeout = timeout || 1; // start the (linear) back-off process
+    return sleep(timeout).then(function () {
+      return submitTransaction(tx, timeout + 5);
+    });
+  });
+}
+```
+
+</CodeExample>
+
+We assume the existence of a sleep implementation similar to the one [here](https://stackoverflow.com/a/39914235). Be sure to integrate backoff into your retry mechanism. In our example error-handling code above, we implement a simple linear backoff, but there are [plenty of recommendations](https://backoff-utils.readthedocs.io/en/latest/strategies.html#why-are-backoff-strategies-useful) for various other strategies. Backoff is important both for maintaining performance and avoiding rate-limiting issues.
+
+### Example: Invalid Sequence Numbers
+
+These errors typically occur when you have an outdated view of an account. This could be because multiple devices are using this account, you have concurrent submissions happening, or other reasons. The solution is relatively simple: retrieve the account details and try again with an updated sequence number.
+
+<CodeExample>
+
+```js
+// suppose `account` is an outdated `AccountResponse` object
+let tx = sdk.TransactionBuilder(account, ...)/* etc */.build();
+server.submitTransaction(tx).catch(function (error)) {
+  if (error.response && error.status == 400 && error.extras &&
+      error.extras.result_codes.transaction == sdk.TX_BAD_SEQ) {
+    return server.loadAccount(account.accountId())
+      .then(function (response) {
+        let tx = sdk.TransactionBuilder(response, ...)/* etc */.build()
+        return server.submitTransaction(tx);
+      });
+  }
+  // ...other error conditions...
+}
+```
+
+</CodeExample>
+
+Despite the solution’s simplicity, things can go wrong fast if you don’t understand why the error occurred.
+
+Suppose you submit transactions from multiple places in your application simultaneously, and your user spammed the Send Payment button a few times in their impatience. If you send the exact same payment transaction for each tap, naturally, only one will succeed. The others will fail with an invalid sequence number (`tx_bad_seq`), and if you resubmit blindly with an updated sequence number (as we do above), these payments will also succeed, resulting in more than one payment being made when only one was intended. So **be very careful when resubmitting transactions** that have been modified to work around an error.
+
+## Managing specific Errors
+
+Here, we will cover specific errors commonly encountered during transaction submission and direct you to the appropriate resolution. We’ll start with a table of common errors and their codes and descriptions, then dive deeper into some specific ones.
```

Comment:
```suggestion
Here, we will cover specific errors commonly encountered during transaction submission and direct you to the appropriate resolution.
```

It looks like we don't have deeper dives, so either add cross-references back up to the examples or omit this sentence :+1:

Reply:
Ah nice catch. I think I removed this deep dive stuff but forgot to refactor this sentence.